src/smallstack.rs

```diff
@@ -92,33 +92,31 @@ impl<'a> VM<'a> {
         &["carry", "invert"] => carry = !carry,
 
         &["call", "primitive", "say"] => if let Some(Expr::Str(arg)) = stack.pop() {
-          println!("hey {}", arg);
+          println!("{}", arg);
         } else {
           panic!("VM error: incorrect arguments to `say`");
         },
 
+        &["call", "primitive", "concat"] => if let
+            (Some(Expr::Str(arg1)), Some(Expr::Str(arg2)))
+            =
+            (stack.pop(), stack.pop()) {
+          stack.push(Expr::Str(arg1 + &arg2));
```

Comment:
I am not sure we should tackle this as a global hash of primitive functions. Maybe we could first tackle modules (as hashes of functions), then introduce the notion of a "native module" which would be implemented in Rust within the VM?